server/client.go

```diff
@@ -1883,7 +1883,7 @@ func (c *client) processConnect(arg []byte) error {
 			}
 		}
 		if ncs != _EMPTY_ {
-			c.ncs.Store(fmt.Sprintf("%s - %q", c, ncs))
+			c.ncs.CompareAndSwap(nil, fmt.Sprintf("%s - %q", c, ncs))
```

Comment:
This Go 1.19 only?

Reply:
Looks like built-in since Go 1.17: https://pkg.go.dev/sync/atomic#Value.CompareAndSwap

Reply:
Does this change pass the fuzz test?

Reply:
yes this would fix it